### PR TITLE
Add automatic detection on 'use' command

### PR DESCRIPTION
**Pull request by Fabianoshz**
## Why? 🤔

https://github.com/cunymatthieu/tgenv/issues/20

## What? :hammer_and_wrench:

Add the ability to automatically use the version file on the current directory.

### CODE REVIEW

**Comment by Fabianoshz** [comment]
This picks up the `.terragrunt-version` file if exists, I've followed the same logic of the [tgenv-install](https://github.com/tgenv/tgenv/blob/main/libexec/tgenv-install#L10-L17) file:

```
if [ -z "${1}" ]; then
  version_file="$(tgenv-version-file)"
  if [ "${version_file}" != "${TGENV_ROOT}/version" ]; then
    version_requested="$(cat "${version_file}" || true)"
  fi
else
  version_requested="${1}"
fi
```

Ideally we could make a helper that finds the version file in the current directory or above, then we could change both `install` and `use` to use the same helper.

**Comment by Fabianoshz** [comment]
Added latest as the default option, not sure if is good.

**Comment by iokiwi** [comment]
Apologies I'm going off grid for a week starting tomorrow so I may not get arround to getting this merged for a few days.

While I'm not opposed to bringing tgenv behaviour in line with tfenv I don't really understand the use case here but I think I see a problem none the less.

With tfenv, if a `.terraform-version` file exists then tfenv will automatically install (if not already installed) and implicitly switch to the version specified in the file to execute your terraform command.

So at least with tfenv, I don't think you should need to explicitly run `tfenv use` in your script to switch to the terraform version specified in `.terraform-version` as it should be picked up by default anyways right?

According to our own docs, tgenv should behave the same, however, If I could now draw your attention to https://github.com/cunymatthieu/tgenv/issues/39 that doesn't appear to be the case.

So in summary, my thoughts are.

 * You should be able to get the behaviour you desire by doing `TGENV_AUTO_INSTALL=true terragrunt <command>`
 * We should fix https://github.com/cunymatthieu/tgenv/issues/39 in our own fork, after which that behaviour becomes the default and you can simply run `terragrunt <command>`
 * Then we should circle back to this this pull request and see where we stand.

**Review by iokiwi — APPROVED** [review]
It looks okay to me. 

```
➜  ~ tgenv list
* 0.42.5 (set by /Users/simon/.tgenv/version)
  0.42.3
  0.42.0
  0.30.6
  0.30.3
  0.30.1
  0.0.1
```

Uses available version listed in .terragrunt-version
```
➜  .tgenv git:(automatic-detection-on-use) ✗ tgenv use
[INFO] Version not specified, looking for version file
[INFO] Switching to v0.30.1
[INFO] Switching completed
```

Does not use version that is not installed.
```
➜  .tgenv git:(automatic-detection-on-use) ✗ tgenv use
tgenv: tgenv-use: [ERROR] No installed versions of terragrunt matched '0.30.4'
```

**Review by weyderfs — APPROVED** [review]

**Comment by weyderfs** [comment]
@iokiwi, First I hope that you had a good new year's eve and vacation, thanks for share your opinion and test the code too. I understand what your mean in the comments above.

The intention in this pull request it's only enable the _tgenv_ `use` command  installing the version set in the `.terragrunt-version` file. Yes this doesn't make sense, but it's useful workaround to lazy guys, such a shortcut 🤣🤣🤣🤣.

But you're right, we need to fix about the `TGENV_AUTO_INSTALL` when is `true`, I'm suffer sometimes with this bug.

**Comment by Fabianoshz** [comment]
Hey guys, thanks for the reviews!
